M/MKL_Headers/build_tarballs.jl

```diff
@@ -2,54 +2,69 @@
 # `julia build_tarballs.jl --help` to see a usage message.
 using BinaryBuilder, Pkg
 
+# Note: 2024.0.0 has already been built, but 2023.2.0 is being built to fix a bug in the packaging where the CMake/pkgconfig scripts were missing.
+# When updating to the next 2024.x.x release, the following must be done again (in addition to updating the sources):
+# * Bump the Julia compat to 1.6
+# * Remove the macos x86_64 platform and source files
```

Comment:
We can also revert the merge commit (with `[skip ci]`) after this PR is merged, no?

Reply:
We can do it that way, I hadn't thought of just reverting it right away.